app/src/pages/inside/settingsPage/generalTab/generalTab.jsx

```diff
@@ -0,0 +1,275 @@
+import React, { Component } from 'react';
+import classNames from 'classnames/bind';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { reduxForm, formValueSelector } from 'redux-form';
+import { connect } from 'react-redux';
+import { Input } from '../../../../components/inputs/input';
+import styles from './generalTab.scss';
+import { InputDropdown } from '../../../../components/inputs/inputDropdown';
+import { InputBigSwitcher } from '../../../../components/inputs/inputBigSwitcher';
+import { BigButton } from '../../../../components/buttons/bigButton';
+import { timeStorageOptions, timeTimeoutOptions } from './timeOptions';
+import { FieldProvider } from '../../../../components/fields/fieldProvider';
+import { InputRadio } from '../../../../components/inputs/inputRadio';
+
+const cx = classNames.bind(styles);
+const cellClasses = cx('gridCell', 'centered');
+const cellClassesSubmit = cx('gridCell', 'submitButton');
+const cellClassesFormField = cx('gridCell', 'formField');
+const cellClassesColumn = cx('gridCell', 'column');
+const cellClassesEmpty = cx('gridCell', 'empty');
+
+const selector = formValueSelector('testRunner');
+
+@reduxForm({
+  form: 'testRunner',
+})
+@connect((state) => {
+  const autoAnalysisEnabled = selector(state, 'autoAnalysis');
+  return {
+    autoAnalysisEnabled,
+  };
+})
+export class GeneralTab extends Component {
+  static defaultProps = {
```

Comment:
In accordance to project code style, we describe propTypes firstly, and then defaultProps.